### PR TITLE
fix  #18776: use XML fragment serialization for innerHTML in XML documents

### LINKED ISSUE #18776: Use XML fragment serialization for innerHTML in XML documents
Per https://w3c.github.io/DOM-Parsing/#serializing and https://github.com/servo/servo/blob/8dece5e74e32223d968648080bb371f3be7fbd7c/components/script/dom/element.rs#L2050-L2054 we don't support this yet.

### PR DESCRIPTION
**Pull request by tigercosmos**
I am not sure whether my solution is in the right way.
@jdm Can you give some advises?

---

- [X] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #18776 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18888)

### CODE REVIEW

**Comment by jdm** [comment]
That looks like the right idea to me.
@bors-servo: try

**Comment by tigercosmos** [comment]
It will fail at https://github.com/servo/servo/blob/master/tests/wpt/web-platform-tests/domparsing/innerhtml-03.xhtml.

Keep working on it.

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /domparsing/innerhtml-03.xhtml:
  └ PASS [expected FAIL] innerHTML in XHTML

  ▶ Unexpected subtest result in /domparsing/innerhtml-03.xhtml:
  └ PASS [expected FAIL] innerHTML in XHTML 1

  ▶ Unexpected subtest result in /domparsing/innerhtml-03.xhtml:
  └ PASS [expected FAIL] innerHTML in XHTML 2

  ▶ Unexpected subtest result in /domparsing/innerhtml-03.xhtml:
  └ PASS [expected FAIL] innerHTML in XHTML 7
```
This is exactly the result that we are hoping for :) Please remove the corresponding ini files in tests/wpt/metadata/domparsing/, since the tests are no longer expected to fail.

**Comment by jdm** [comment]
Actually, I think I was overeager. So we pass some of the tests in that file now, which is an improvement. We should remove those specific lines from https://dxr.mozilla.org/servo/source/tests/wpt/metadata/domparsing/innerhtml-03.xhtml.ini?q=path%3Ainnerhtml-03.xhtml.ini&redirect_type=single#3 rather than the entire file.

**Comment by jdm** [comment]
I guess https://github.com/servo/servo/blob/master/tests/wpt/web-platform-tests/domparsing/innerhtml-03.xhtml#L32-L36 is the only test that we still fail in that file.

**Comment by tigercosmos** [comment]
Actually still remain two test:
https://github.com/servo/servo/blob/master/tests/wpt/web-platform-tests/domparsing/innerhtml-03.xhtml#L32-L41
```
  [innerHTML in XHTML 3]
    expected: FAIL

  [innerHTML in XHTML 4]
    expected: FAIL
```

**Comment by tigercosmos** [comment]
I have no idea how to fix these two.

**Comment by jdm** [comment]
We can investigate them separately from this PR.

**Comment by jdm** [comment]
Let's fix outerHTML in the same way, too.

File: tests/wpt/metadata/html/syntax/serializing-xml-fragments/outerHTML.html.ini [hunk 1]
```diff
@@ -1,296 +1,11 @@
...
-  [Node for var]
-    expected: FAIL
-
-  [Node for video]
-    expected: FAIL
-
-  [Node for data]
-    expected: FAIL
-
   [Node for area]
```
Comment (tigercosmos) [inline tests/wpt/metadata/html/syntax/serializing-xml-fragments/outerHTML.html.ini]: I don't know why these remain to fail.

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Comment by tigercosmos** [comment]
how to see the error log?

**Comment by jdm** [comment]
Clicking `filtered-wpt-errorsummary.log` shows the unexpected test results, and `stdio` for the `test-wpt` job shows the regular output. More passing tests!
```

  ▶ Unexpected subtest result in /html/semantics/scripting-1/the-template-element/additions-to-serializing-xhtml-documents/outerhtml.html:
  └ PASS [expected FAIL] Template contents should be serialized instead of template element if serializing template element

  ▶ Unexpected subtest result in /html/semantics/scripting-1/the-template-element/additions-to-serializing-xhtml-documents/outerhtml.html:
  └ PASS [expected FAIL] Template contents should be serialized instead of template element if serializing template element. Test nested template

  ▶ Unexpected subtest result in /html/semantics/scripting-1/the-template-element/additions-to-serializing-xhtml-documents/outerhtml.html:
  └ PASS [expected FAIL] Template contents should be serialized instead of template element if serializing template element. Test serializing whole document
```

**Comment by tigercosmos** [comment]
Where is `/_mozilla/` folder? Seems not in `tests`?
```  
▶ FAIL [expected PASS] /_mozilla/css/border_radius_elliptical_a.html
  └   → /_mozilla/css/border_radius_elliptical_a.html 1a25abe342c5059bd0971151cd815ddb48bcfde4
/_mozilla/css/border_radius_elliptical_ref.html ef32bea13c7811e9092d6f334643fda26a617ef6
Testing 1a25abe342c5059bd0971151cd815ddb48bcfde4 == ef32bea13c7811e9092d6f334643fda26a617ef6
```

**Comment by jdm** [comment]
That one is a known intermittent failure, so you should ignore it (which is why it wasn't included in the list of filtered-wpt-errorsummary.log. However, those tests live in tests/wpt/mozilla/tests/, and the ini files are in tests/wpt/mozilla/meta.

**Comment by KiChjang** [comment]
Squash!

**Comment by tigercosmos** [comment]
@KiChjang  Shall we separate `innerHTML` and `outerHTML` to different commits?

**Comment by KiChjang** [comment]
Oh, I misread. I thought they're the same thing... I would still squash, but I don't mind either way.

Thanks for your work!

@bors-servo r=jdm

**Pull request merged**